components/builder-router/src/server.rs

```diff
@@ -278,11 +278,7 @@ impl Server {
 
     fn select_shard(&mut self) -> u32 {
```

Comment:
I did some digging. The thing is that the protobuf library does an `unwrap_or(0)` when you call `get_hash()` on the route_info. What you want to do here is see if hash is `Some` or `None` and not pull the value out

```rust
    fn select_shard(&mut self) -> u32 {
        if self.envelope.route_info().has_hash() {
            (self.envelope.route_info().get_hash() % SHARD_COUNT as u64) as u32
        } else {
            (self.rng.gen::<u64>() % SHARD_COUNT as u64) as u32
        }
    }
```